clio-server/src/test/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAOSpec.scala

```diff
@@ -0,0 +1,74 @@
+package org.broadinstitute.clio.server.dataaccess
+
+import org.broadinstitute.clio.server.dataaccess.elasticsearch.{
+  DocumentMock,
+  ElasticsearchIndex
+}
+import org.broadinstitute.clio.server.dataaccess.elasticsearch.Elastic4sAutoDerivation._
+import org.broadinstitute.clio.server.dataaccess.util.ClioUUIDGenerator
+
+import com.sksamuel.elastic4s.circe._
+import io.circe.parser._
+import org.scalatest.{AsyncFlatSpec, Matchers}
+
+import scala.collection.JavaConverters._
+
+import java.nio.file.Files
+
+class PersistenceDAOSpec extends AsyncFlatSpec with Matchers {
+  behavior of "PersistenceDAO"
+
+  val index: ElasticsearchIndex[DocumentMock] = DocumentMock.index
+
+  it should "initialize top-level storage for indexed documents" in {
+    val dao = new JimfsPersistenceDAO()
+    val wgsPath = dao.rootPath.resolve(index.rootDir)
+
+    Files.exists(wgsPath) should be(false)
+    Files.isDirectory(wgsPath) should be(false)
+
+    for {
+      _ <- dao.initialize(index)
+    } yield {
+      Files.exists(wgsPath) should be(true)
+      Files.isDirectory(wgsPath) should be(true)
+    }
+  }
+
+  it should "write metadata updates to storage using the UUID" in {
+    val dao = new JimfsPersistenceDAO()
+    val document = DocumentMock(
```

Comment:
Is this really a mock? It looks like a valid document that is only used in a test, not a mock.

To make our test code simpler, the test document could also have a constructor that creates it with dummy data, to avoid the need to put that code here.